pypsa/components.py

```diff
@@ -391,6 +403,10 @@ def _build_dataframes(self):
                 columns=static_dtypes.index,
             )
 
+            if component in self.geo_components:
```

Comment:
@pz-max  this is the code isn't backward compatible because we create dataframes for bus links when initializing a network, and here we convert them to GeoPandas dataframes.

Reply:
We can abstract this into a separate method, allowing us to easily convert between standard dataframes and GeoPandas as needed.

Reply:
Got it! :+1: